frontend/src/app/Settings/Facility/Components/FacilityInformation.tsx

```diff
@@ -1,83 +1,93 @@
 import React from "react";
 import { stateCodes } from "../../../../config/constants";
 import Dropdown from "../../../commonComponents/Dropdown";
-import TextInput from "../../../commonComponents/ManagedTextInput";
+import TextInput from "../../../commonComponents/TextInput";
```

Comment:
A `ManagedTextInput` was a copy of a `TextInput` with changes that didn't seem that divergent. The main thing is that inputs need to have a `name` so that the event handler knows which field to update, and I think i've done that successfully.

Reply:
`ManagedTextInput` has a couple of advantages over `TestInput`: 
- Cleaner separation of form logic and input rendering details. the form should have to know what element the Input components use in their rendering. This allows for easier refactoring of the internals of an input container and easier testing of form logic
- Using the name field loses strict typing. The values of the name attribute are no longer compared to the keys of that Facility type. If we are to update the Facility type field from `name` to `testing_lab` Typescript will not fail.

Reply:
This is kind of a choose-your-poison thing I suspect. With the `ManagedTextInput` you have to create a custom event handler for every input which gets really messy and verbose as you can see from the diff. If the strict typing is a problem we can do `TextInput<ValueType extends string>` and specify the type when invoking, but we'd definitely be having to wrap any third-party component because none of them do strict typing like that AFAIK. The middle ground we used in AG is to have a `nameForKey` wrapper that's used for form field names, so you'd say `nameForKey<Facility>("name")` and if there is no facility field `name` then you'll get an error. To do that we need TS types for all our records and I don't think we have those yet.